### PR TITLE
db_infos

### PR DESCRIPTION
**Pull request by yinchimaoliang**
Change gt_database and dbsampler.

### CODE REVIEW

File: mmdet3d/datasets/pipelines/dbsampler.py [hunk 1]
```diff
@@ -94,7 +94,9 @@ def __init__(self,
                  rate,
                  prepare,
                  sample_groups,
-                 classes=None):
+                 classes=None,
+                 load_dim=4,
+                 use_dim=[0, 1, 2, 3]):
```
Comment (Tai-Wang) [inline mmdet3d/datasets/pipelines/dbsampler.py:99]: Can we support int type for use_dim like loading.py?

File: tools/data_converter/create_gt_database.py [hunk 2]
```diff
@@ -175,8 +175,10 @@ def create_groundtruth_database(dataset_class_name,
             dict(type='LoadPointsFromFile', load_dim=5, use_dim=5),
             dict(
                 type='LoadPointsFromMultiSweeps',
-                sweeps_num=10,
-            ),
+                sweeps_num=9,
```
Comment (ZwwWayne) [inline tools/data_converter/create_gt_database.py:178]: We may still use `sweeps_num=10` because this best optimize the training settings.

**Review by ZwwWayne — APPROVED** [review]

File: mmdet3d/datasets/pipelines/dbsampler.py (hunk 1 above)
Comment (ZwwWayne) [inline mmdet3d/datasets/pipelines/dbsampler.py:99]: Update docstrings accordingly.

File: mmdet3d/datasets/pipelines/dbsampler.py [hunk 3]
```diff
@@ -103,6 +108,10 @@ def __init__(self,
         self.classes = classes
         self.cat2label = {name: i for i, name in enumerate(classes)}
         self.label2cat = {i: name for i, name in enumerate(classes)}
+        self.load_dim = load_dim
+        if isinstance(use_dim, int):
+            use_dim = list(range(use_dim))
+        self.use_dim = use_dim
 
         with open(info_path, 'rb') as f:
```
Comment (ZwwWayne) [inline mmdet3d/datasets/pipelines/dbsampler.py:116]: Use mmcv to load the pkl file.

File: mmdet3d/datasets/pipelines/dbsampler.py [hunk 4]
```diff
@@ -245,7 +254,9 @@ def sample_all(self, gt_bboxes, gt_labels, img=None):
                     self.data_root,
                     info['path']) if self.data_root else info['path']
                 s_points = np.fromfile(
-                    file_path, dtype=np.float32).reshape([-1, 4])
+                    file_path,
```
Comment (ZwwWayne) [inline mmdet3d/datasets/pipelines/dbsampler.py:257]: Support to use `fileclient`. Therefore, we may consider using `LoadPointsFromFile` here.

File: mmdet3d/datasets/pipelines/dbsampler.py [hunk 5]
```diff
@@ -103,9 +110,11 @@ def __init__(self,
         self.classes = classes
         self.cat2label = {name: i for i, name in enumerate(classes)}
         self.label2cat = {i: name for i, name in enumerate(classes)}
+        self.points_loader = mmcv.build_from_cfg(points_loader, PIPELINES)
 
-        with open(info_path, 'rb') as f:
-            db_infos = pickle.load(f)
+        db_infos = mmcv.load(info_path)
+        # with open(info_path, 'rb') as f:
```
Comment (ZwwWayne) [inline mmdet3d/datasets/pipelines/dbsampler.py:116]: delete comments in lines 116-117

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**